src/code_808417C.c

```diff
@@ -0,0 +1,26 @@
+#include "global.h"
+#include "constants/ability.h"
+#include "constants/iq_skill.h"
+#include "dungeon_entity.h"
+#include "dungeon_pokemon_attributes.h"
+#include "dungeon_random.h"
+#include "dungeon_util.h"
+
+s32 sub_808417C(struct DungeonEntity *param_1, s16 *param_2, u8 param_3)
```

Comment:
Based on the presence of Self Curer and Natural Cure, this function gets the number of turns a status lasts.
`param_1` is the Pokémon getting statused.
`param_2` is a size-2 array (or a struct) with the min and max possible turns for the status.
`param_3` is a `bool8` determining whether to factor in Self Curer and Natural Cure.

Reply:
Renamed to `CalculateStatusTurns`